content/docs/practices/the_zen.md

```diff
@@ -0,0 +1,118 @@
+
+[RED](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/), [USE](http://www.brendangregg.com/usemethod.html) and [The Four Golden Signals](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/#xref_monitoring_golden-signals) are known frameworks to get you started.
+
+This doesn't necessarily mean that you should ignore causal metrics. Have them, spread them liberally.
+
+    Use causal metrics to answer why something is broken.
+
+## Labels are the new hierarchies
+
+**Labels are the new hierarchies**, yet more powerful and flexible. Labels are what make Prometheus strong. Using labels, one can group and aggregate measurements afterwards. Slice and dice using Labels. Remember [Instrument first, ask questions later](#instrument-first-ask-questions-later), provide much context as possible.
+
+However, you have to *use labels with care*. The reasons will explain in subsequent rules.
+
+## Avoid missing metrics
+
+Time series that are not present until something happens are difficult to deal with. To avoid this, export 0 (or NaN if 0 would be misleading) for any time series you know may exist in advance. You have to initialize your metrics with the zero values to prevent broken dashboards and misfiring alerts. For more detailed explanation check out [Existential issues with metrics](https://www.robustperception.io/existential-issues-with-metrics).
+And also remember, labels create timeseries, so same goes for your labels. Your client libraries can't know what labels you would have. Initialize your metrics with the labels that you would have.
+
+## Cardinality Matters
+
+Every unique set of labels create a new timeseries. Use labels with care watch out what you put into your labels. Avoid cardinality explosion, unbounded labels will blow up Prometheus. And keep in mind that labels are multiplicative. You will have multiple labels, multiple target labels and targets.
+
+    Prometheus performance almost always comes down to one thing: label cardinality.
+
+Always remember that [Cardinality is key](https://www.robustperception.io/cardinality-is-key).
+
+## Naming is hard
+
+One of the hardest problem in computer science. Metric name should only have a single meaning within a single job. And you should always guard against metric name collisions among the jobs on your system.
+
+*Respect conventions over preferences*. Conventions are no one's favorite, yet conventions are everyone's favorite. See the documentation on [Naming](https://prometheus.io/docs/practices/naming/) conventions for the nitty-gritty details.
+
+One other benefit of following the conventions is that you don't need to reinvent the wheel. You can benefit from the tools that's out there, such as [Monitoring Mixins](https://monitoring.mixins.dev/).
+
+---
+
+## Counters rule and gauges suck
```

Comment:
This is overly harsh, lots of things can only be reasonably done with gauges - and some things require both.

Reply:
It's not a nuanced statement, it's a proverb!

It refers to not using gauges for something that can be represented as a counter. That's explained in the text below.